src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1181,13 +1184,17 @@ CheckCitusVersion(int elevel)
  * this function logs an error with the specified elevel and returns false,
  * otherwise it returns true.
  */
-static bool
+bool
 CheckAvailableVersion(int elevel)
 {
 	char *availableVersion = NULL;
 
-	Assert(CitusHasBeenLoaded());
-	Assert(EnableVersionChecks);
+	if (citusVersionKnownCompatible ||
```

Comment:
On second thought, I think this does no affect ALTER EXTENSION, but it causes exit early for CREATE EXTENSION for the same reason. We perform this check before extension creation. I removed this checks.